### PR TITLE
Changed all instances of k-dev@ and kubernetes-dev@ to dev@kubernetes.io

### LINKED ISSUE #6275: [ML Migration] Post migration: kdev mailing list
Umbrella #5877 

Things to do once the group has been created.

1. run a cs.k8s.io on k-dev@ and kubernetes-dev@ and change all respective instances to dev@kubernetes.io (https://groups.google.com/a/kubernetes.io/group/dev)
1.5 also google it to see what comes up and make the changes to the front page search results
  -pull for community repo #6311 
  -need to look at all repos
  -need to look at google search results 
  
2. have leads change all docs starting most notably with meeting invites and any open work collaboration docs
 -have leads also check that contributors@ is the owner
 -if contributor@ is already an owner, we can help add to docs 

3. Forward any active conversations to the new list 

what else?

### PR DESCRIPTION
**Pull request by chetak123**
**Which issue(s) this PR fixes**:

Fixes #6275

### CODE REVIEW

**Comment by parispittman** [comment]
/assign
/hold

thank you so much!! I'll review asap but it will take me a bit. will try and have a review done by this Thursday.

**Review by Debanitrkl — APPROVED** [review]
Went through the changes looks good to me
/ok-to-test

**Comment by cblecker** [comment]
Please review https://github.com/kubernetes/community/pull/6311#issuecomment-1001575036 and remove any changes to documents in these paths. Please also remove changes to documents in the `archive/` folder. Thanks!

File: community-membership.md
Comment (cblecker) [inline community-membership.md:266]: This link needs to be changed to the new destination

File: communication/meeting-notes-archive/201801-201807_Community_Meeting_Minutes.md
Comment (cblecker) [inline communication/meeting-notes-archive/201801-201807_Community_Meeting_Minutes.md:1296]: We don't need to change historical documents

**Review by cblecker — CHANGES_REQUESTED** [review]

File: community-membership.md
Comment (chetak123) [reply community-membership.md:266]: Hey @cblecker it would really helpful if you can please provide link for new forum destination as I am not able to find it for dev@kubernetes.io

Comment (parispittman) [inline community-membership.md:266]:
```suggestion
[dev@kubernetes.io]: https://groups.google.com/a/kubernetes.io/group/dev
```

**Review by parispittman — CHANGES_REQUESTED** [review]

**Comment by parispittman** [comment]
Thank you @chetak123 for your work on this! I left the appropriate link; once that is updated, I'll release the hold and we can merge. :)

File: sig-contributor-experience/meeting-notes-archive/2018-meeting-notes.md
Comment (parispittman) [inline sig-contributor-experience/meeting-notes-archive/2018-meeting-notes.md:2854]: this file does not need to be changed as its an archive so we should keep the history

File: sig-contributor-experience/meeting-notes-archive/2019-meeting-notes.md
Comment (parispittman) [inline sig-contributor-experience/meeting-notes-archive/2019-meeting-notes.md:2827]: this file does not need to be changed as its an archive so we should keep the history

File: sig-contributor-experience/surveys/2019/contribex-survey-2019.md
Comment (parispittman) [inline sig-contributor-experience/surveys/2019/contribex-survey-2019.md:16]: this file does not need to be changed as its an archive so we should keep the history  (same with the other survey csv file that you have hooked)

File: sig-contributor-experience/surveys/k8s_survey_analysis/prepare_2018.py
Comment (parispittman) [inline sig-contributor-experience/surveys/k8s_survey_analysis/prepare_2018.py:28]: this file does not need to be changed as its an archive so we should keep the history

File: sig-docs/migrated-from-wiki/roadmap-docs.md
Comment (parispittman) [inline sig-docs/migrated-from-wiki/roadmap-docs.md:11]: this file does not need to be changed as its an archive so we should keep the history

**Review by parispittman — CHANGES_REQUESTED** [review]
sorry about that - missed 5 files that should not be changed as they are archives / should be kept for historical reasons. after that, we are really ready to merge :)

File: contributors/devel/sig-node/e2e-node-tests.md
Comment (cblecker) [inline contributors/devel/sig-node/e2e-node-tests.md:50]: This link needs to change to the new forum

File: events/events-team/best-practices.md
Comment (cblecker) [inline events/events-team/best-practices.md:65]: line120 needs to change with this link

**Review by cblecker — CHANGES_REQUESTED** [review]

**Comment by parispittman** [comment]
thanks so much for sticking through this!
/hold cancel
/lgtm

will wait for Christoph or another approver

File: sig-contributor-experience/charter.md
Comment (cblecker) [inline sig-contributor-experience/charter.md:59]:
```suggestion
    - We will go to each lead, their mailing lists, slack channel, and/or their update meetings and ask for feedback and a [lazy consensus] process. We will follow up with a post to [dev@kubernetes.io] mailing list
```

File: sig-contributor-experience/charter.md
Comment (cblecker) [inline sig-contributor-experience/charter.md:65]:
```suggestion
        - [dev@kubernetes.io] with the GitHub issue link including the subject [NOTICE]: $announcement
```

**Review by cblecker — COMMENTED** [review]
Last couple ones, but as we want to get this in, I'm just gonna commit these and merge.

Thanks @chetak123 !

File: events/elections/2020/README.md
Comment (cblecker) [inline events/elections/2020/README.md:73]:
```suggestion
If you want to stand for election, send an email to kubernetes-dev@googlegroups.com
```

File: events/elections/2021/README.md
Comment (cblecker) [inline events/elections/2021/README.md:153]:
```suggestion
2. After creating the issue, send an email to kubernetes-dev@googlegroups.com
```

**Review by cblecker — COMMENTED** [review]
Oh, these two as well are from elections that have already happened

**Comment by chetak123** [comment]
Hey @parispittman I wanted to participate in this [mentorship program](https://github.com/kubernetes/community/issues/6665)  and for that I require github org membership. I have [contributed to kubernetes-sigs in past](https://kubernetes.slack.com/archives/C09R23FHP/p1654108446287059) and would love to do so in future. 
It would be really helpful if you can please one of my sponsor for getting  the org membership so that I can meet the requirement for this program.
Thanks

**Pull request merged**
